### PR TITLE
Draft: Add config option for logging

### PR DESCRIPTION
**Pull request by jardon**
This is a PR I threw together for supporting the configuration of the logging directory.  I am by no means a C# developer so if you think this is garbage, you won't hurt my feelings.  If this is something that would be useful to the project, I can iterate on it to clean it up.

Also, if there's a better way to do this, I am all ears.  I spent a little bit of time working on this because I was playing around with trying to package this differently and dumping logs into the same directory as the binary was becoming problematic.

### CODE REVIEW

**Comment by TSRBerry** [comment]
> dumping logs into the same directory as the binary was becoming problematic

Related: #4707

File: Ryujinx.Common/Logging/Targets/FileLogTarget.cs [hunk 1]
```diff
@@ -12,14 +14,25 @@ public class FileLogTarget : ILogTarget
...
+        private static string _config = GetConfigName();
+
+        // Build a config object, using env vars and JSON providers.
+        private static IConfiguration config = new ConfigurationBuilder()
+            .AddJsonFile(_config)
+            .AddEnvironmentVariables()
+            .Build();
+
+        // Get values from the config given their key and their target type.
+        public static Settings settings = config.GetRequiredSection("Settings").Get<Settings>();
```
Comment (WilliamWsyHK) [inline Ryujinx.Common/Logging/Targets/FileLogTarget.cs:26]: `public static Settings Settings` (notice the `S` is capital letter in the Property name)

File: Ryujinx.Common/Settings.cs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+public sealed class Settings
+{
+    public required string logDir { get; set; }
```
Comment (WilliamWsyHK) [inline Ryujinx.Common/Settings.cs:3]: `public required string LogDirectory` (`l` to `L` and I just tried to make it more readable for the property name.)

File: Ryujinx/linux.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+    "Settings": {
+        "logDir": "/home/jardon/.local/ryujinx"
```
Comment (WilliamWsyHK) [inline Ryujinx/linux.json:3]: Please do not hard-code the user name. Other *nix platform as well.

File: Ryujinx/windows.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+    "Settings": {
+        "logDir": "C:"
```
Comment (WilliamWsyHK) [inline Ryujinx/windows.json:3]: I suppose it is better not to write to C: drive directly.

**Review by WilliamWsyHK — COMMENTED** [review]
I believe for the path, we might use the current `logs` folder under the current Ryujinx config folder for this purpose.
If it is portable, `logs` folder should be created under the portable folder.

File: Ryujinx/linux.json (hunk 3 above)
Comment (jardon) [reply Ryujinx/linux.json:3]: This is a draft :) Those were just some values to prove that it worked.

**Pull request closed without merging**